scripts/sct_label_vertebrae.py

```diff
@@ -391,26 +396,29 @@ def main(args=None):
                  'NearestNeighbor'),
                 verbose=verbose,
                 is_sct_binary=True,
-               )
+                )
         # get z value and disk value to initialize labeling
+        # After resampling to match template resolution
+        sct.run('sct_resample -i labelz_straight.nii.gz -mm 0.5 -x nn -o labelz_straight_r.nii.gz')
         sct.printv('\nGet z and disc values from straight label...', verbose)
-        init_disc = get_z_and_disc_values_from_label('labelz_straight.nii.gz')
+        init_disc = get_z_and_disc_values_from_label('labelz_straight_r.nii.gz')
         sct.printv('.. ' + str(init_disc), verbose)
 
         # denoise data
         if denoise:
             sct.printv('\nDenoise data...', verbose)
-            sct.run(['sct_maths', '-i', 'data_straightr.nii', '-denoise', 'h=0.05', '-o', 'data_straightr.nii'], verbose)
+            sct.run(['sct_maths', '-i', 'data_straightr.nii', '-denoise', 'h=0.05', '-o', 'data_straightr.nii'],
+                    verbose)
 
         # apply laplacian filtering
         if laplacian:
             sct.printv('\nApply Laplacian filter...', verbose)
             sct.run(['sct_maths', '-i', 'data_straightr.nii', '-laplacian', '1', '-o', 'data_straightr.nii'], verbose)
 
         # detect vertebral levels on straight spinal cord
-        init_disc[1]=init_disc[1]-1
```

Comment:
Off-by-one cause?

Reply:
I think so, let me check and get back to you on that.